packages/ui/Button/Button.js

```diff
@@ -167,10 +175,27 @@ const ButtonBase = kind({
 			pressed,
 			selected
 		}, size),
-		icon: ({css, icon, iconComponent: Icon, size}) => {
-			return (typeof icon === 'string' && Icon) ? (
-				<Icon size={size} className={css.icon}>{icon}</Icon>
-			) : icon;
+		icon: ({css, icon, iconComponent, size}) => {
+			if (icon == null || icon === false) return;
+
+			// Establish the base collection of props for the moost basic `iconComponent` type, an
+			// HTML element string.
+			const props = {
+				className: css.icon,
+				component: iconComponent
+			};
+
+			// Add in additional props that any Component supplied to `iconComponent` should be
+			// configured to handle.
+			if (typeof iconComponent !== 'string') {
```

Comment:
@Djspaceg - Would like to talk through this change a bit more together. We're creating an exception here -- `iconComponent` can be a string and thereby opts out of receiving the custom props -- and I want to understand the conditions for the exception better.

Reply:
ComponentOverride automatically excludes props that are in all lowercase? I didn't realize it did any props augmentation. That's really cool. We can then refactor this whole conditional then.

Reply:
> ComponentOverride automatically excludes props that are in all lowercase?

No. Sorry for the confusion. `ComponentOverride` passes everything through. My point was that this is the only place where we attempt to process props differently when overriding. Everywhere else we always pass all the documented props through.

Reply:
> `iconComponent` can be a string and thereby opts out of receiving the custom props

I was calling out the exception and not stating the current behavior. I see how that was confusing!

Reply:
Ah ok. This seemed like a good value-add, since this code could be really buried in the stack. It didn't seem intuitive to me to have `size` added to a `div`. If a developer creates their own component, and it is just a plain `div`, it will actually be a React component and will get the `size` prop, and because it's a component, I'll have the ability to prune it off, if I don't want to act on it.